packages/autorest.python/autorest/codegen/serializers/__init__.py

```diff
@@ -495,6 +507,42 @@ def _serialize_and_write_metadata(
             namespace_path / Path("_metadata.json"), metadata_serializer.serialize()
         )
 
+    # find root folder where "setup.py" is
+    def _package_root_folder(self, namespace_path: Path) -> Path:
+        return namespace_path / Path("../" * (self.code_model.namespace.count(".") + 1))
+
+    def _serialize_and_write_sample(self, env: Environment, namespace_path: Path):
+        out_path = self._package_root_folder(namespace_path) / Path("generated_samples")
+        for op_group in self.code_model.operation_groups:
+            if self.code_model.options["multiapi"]:
+                api_version_folder = f"{op_group.api_versions[0]}/"
+            else:
+                api_version_folder = ""
+            for operation in op_group.operations:
```

Comment:
ideally we just have the `SampleSerializer` take in the code model, then when we call `.serialize()` on the serializer, we go through all of the operation groups and operations ourselves. That way we keep the main file cleaner, and handle the sample serialization in the serializer

Reply:
@iscai-msft I keep same with mgmt that [one operation_group has one output file](https://github.com/Azure/autorest.python/blob/5a8dd65b4618de1a30a606afb087d50fe624f5bb/packages/autorest.python/autorest/codegen/serializers/__init__.py#L358-L364). `.serialize()` only generate code of one file and it is the current design of autorest.python